src/metabase/util/schema.clj

```diff
@@ -356,20 +356,25 @@
                                                     false)))
     (deferred-tru "value must be a valid JSON string.")))
 
+(def ^:private keyword-or-non-blank-str
+  (s/conditional
+    string?  NonBlankString
+    keyword? s/Keyword))
+
 (def Parameter
   "Schema for a valid Parameter.
   We're not using [metabase.mbql.schema/Parameter] here because this Parameter is meant to be used for
   Parameters we store on dashboard/card, and it has some difference with Parameter in MBQL."
-  (with-api-error-message {:id                         NonBlankString
-                           :type                       (s/conditional
-                                                         string?  NonBlankString
-                                                         keyword? s/Keyword)
+  (with-api-error-message {:id                              NonBlankString
+                           :type                            keyword-or-non-blank-str
+                           (s/optional-key :source)         keyword-or-non-blank-str
+                           (s/optional-key :source-options) Map
```

Comment:
`sectionId` is using camelCase. And I made an explicit enum of the expected values:

```
$ git diff src/metabase/util/schema.clj
diff --git a/src/metabase/util/schema.clj b/src/metabase/util/schema.clj
index ac0971a30b..774a26fec8 100644
--- a/src/metabase/util/schema.clj
+++ b/src/metabase/util/schema.clj
@@ -360,16 +360,17 @@
   "Schema for a valid Parameter.
   We're not using [metabase.mbql.schema/Parameter] here because this Parameter is meant to be used for
   Parameters we store on dashboard/card, and it has some difference with Parameter in MBQL."
-  (with-api-error-message {:id                         NonBlankString
-                           :type                       (s/conditional
+  (with-api-error-message {:id                          NonBlankString
+                           :type                        (s/conditional
                                                          string?  NonBlankString
                                                          keyword? s/Keyword)
                            ;; Allow blank name and slug #15279
-                           (s/optional-key :name)      s/Str
-                           (s/optional-key :slug)      s/Str
-                           (s/optional-key :default)   s/Any
-                           (s/optional-key :sectionId) NonBlankString
-                           s/Keyword                   s/Any}
+                           (s/optional-key :name)       s/Str
+                           (s/optional-key :slug)       s/Str
+                           (s/optional-key :default)    s/Any
+                           (s/optional-key :sectionId)  NonBlankString
+                           (s/optional-key :sourceType) (s/enum "static-list" "card" "all-values")
+                           s/Keyword                    s/Any}
     (deferred-tru "parameter must be a map with :id and :type keys")))
 
 (def ParameterMapping
```

Reply:
I guess I called it sourceType instead of source. I don't have strong feelings either way

Reply:
ah yea, js does not like dash.

And I like `sourceType` better. so let's go with that